src/test/java/uk/gov/justice/digital/job/DataReconciliationJobTest.java

```diff
@@ -50,9 +50,20 @@ void runJobShouldReconcileData() {
     @Test
     @ExpectSystemExitWithStatus(1)
     @SuppressWarnings("java:S2699")
-    void runJobShouldSystemExitWithErrorExitCodeWhenResultIsFailure() {
+    void runJobShouldExitWithErrorCodeWhenConfiguredToAndResultIsFailure() {
         when(dataReconciliationService.reconcileData(sparkSession)).thenReturn(results);
         when(results.isSuccess()).thenReturn(false);
+        when(jobArguments.shouldReconciliationFailJobIfChecksFail()).thenReturn(true);
+
+        underTest.runJob(sparkSession);
+    }
+
+    @Test
+    @SuppressWarnings("java:S2699")
+    void runJobShouldNotExitWithErrorCodeWhenConfiguredNotToButResultIsFailure() {
+        when(dataReconciliationService.reconcileData(sparkSession)).thenReturn(results);
+        when(results.isSuccess()).thenReturn(false);
+        when(jobArguments.shouldReconciliationFailJobIfChecksFail()).thenReturn(false);
 
         underTest.runJob(sparkSession);
```

Comment:
Is it worth asserting that the exit code is not > 0?

Reply:
In this case the job should not exit at all so the test is checking that there is no System.exit. I will make the test name a bit clearer and add a comment